### ISSUE TITLE
Samsung Windfree

### ISSUE THREAD

**Issue by buciutaandrei** [veista/smartthings#82]
Hi!

I encountered a problem with my WindFree AC. More specifically when I create a scene for my AC and I setup it as Cool - windFree preset as soon as it starts the preset switches to off. I can't make it stay on windFree.
The only way I can is by setting the scene to the windFreeSleep preset but that also activates the sleep mode.

Thank you!

**Comment by gacopl** [comment]
Yeah confirm it has been like that from beginning which is odd. Windfeee is available only in heat/cool aka auto which is weird as in original app it's other way around windfeee is not available in auto.

**Comment by buciutaandrei** [comment]
Actually I don't have windFree in Heat/Cool mode, only in Cool. Maybe you have an older version. Still, the windfree preset resets as soon as I trigger the scene. When I configure the scene it works perfectly, only on trigger it changes to "off".

**Comment by veista** [comment]
Hi. Windfree on my model is only available in cool mode. I just assume it is this way in every model. My model does not state the missing of windfree in other modes and errors out if changed in other modes. I think the information is available on OCF side, but smartthings does not offer it. If your model supports windfree on other modes you will have to edit the code.

**Comment by gacopl** [comment]
yeah sorry for the confusion i made it is actually the other way around (proper?) windfree is available in cool but not in heatcool (aka auto) same as in samsung app as auto takes care of everything and switches to windfree when conditions allow for it.

But yes i observe the same issue windfree does not persist (on/off cool) but i guess it's samsung fault as it behaves the same from normal remote and samsung app.

I don't use scene but automations to set it the way i need

**Comment by mchiriciuc** [comment]
> Hi. Windfree on my model is only available in cool mode. I just assume it is this way in every model. My model does not state the missing of windfree in other modes and errors out if changed in other modes. I think the information is available on OCF side, but smartthings does not offer it. If your model supports windfree on other modes you will have to edit the code.

Hi,
My AC supports windfree in Heat mode too. But in HA all presets disappear (off, sleep, quiet - aka windfree, smart, speed) in Heat mode. But they show in Cool, Dry and Fan Only. In Smartthings app the presets are available for heat mode too.
If I can edit the code, can you point to where and what?

Big Thanks, great work!

**Comment by veista** [comment]
Hi,

You can edit the climate.py file.
